### ISSUE TITLE
A Sub-query in FROM clause does not work any more

### ISSUE THREAD

**Issue by tdhung80** [orientechnologies/orientdb#5573]
Hi,

I am migrating from version 1.7.8 to version 2.1.8, I get an error with such query in version 2.1.8
```SQL
SELECT referredBy.size() FROM (FIND REFERENCES #15:4)
```
the error is
```Java
com.orientechnologies.orient.core.sql.OCommandSQLParsingException: Error on parsing command at position #0: Error parsing query: SELECT referredBy.size() FROM (FIND REFERENCES #15:4) Encountered " <FROM> "FROM "" at line 1, column 26.
 Was expecting one of:
 <EOF> 
 <TIMEOUT> ...
 <AS> ...
 <FETCHPLAN> ...
 <LOCK> ...
 <LET> ...
 <NOCACHE> ...
 <PARALLEL> ...
 <UNWIND> ...
 ";" ...
 "," ...
 <AS> ...
 "," ...
```
Remarks, the query is only working if strictSQL is unset. What's the correct syntax in this case when strictSQL is set?

My Best,
Hung Tran

**Comment by sinfex** [comment]
This seems to touch https://github.com/orientechnologies/orientdb/issues/4267

```
orientdb {db=test}> FIND REFERENCES #13:0
```

> Found [{rid:#13:0,referredBy:[1]}] in 0.018000 sec(s).

Any reason this does not work?

```
orientdb {db=test}> SELECT referredBy FROM (FIND REFERENCES #13:0)
```

> Error: com.orientechnologies.orient.core.sql.OCommandSQLParsingException: Error on parsing command at position #0: Error parsing query

**Comment by StarpTech** [comment]
The `find references` statement cannot be used in sub querys. Try:

```sql
begin
let $t0 = FIND REFERENCES #15:4
let $t1 = SELECT referredBy.size() FROM $t0
return $t1
```

**Comment by luigidellaquila** [comment]
Hi guys,

we just pushed a fix in 2.1.x branch, now we are porting it to develop.
The fix will be available in v 2.1.10

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Pushed fix no develop branch as well.

Thanks

Luigi
